platforms/allwinner-d1/core/src/lib.rs

```diff
@@ -278,9 +297,12 @@ impl D1 {
 
     /// DMAC ISR handler
     ///
-    /// At the moment, we only service the Channel 0 interrupt,
-    /// which indicates that the serial transmission is complete.
+    /// At the moment, we service the interrupts on the following channels:
+    /// * Channel 0: UART0 TX
+    /// * Channel 1: SPI1 TX
+    /// * Channel 2: TWI0 driver TX
     fn handle_dmac() {
+        let _isr = kernel::isr::Isr::enter();
```

Comment:
Might want a comment to explain what this is doing (swapping tracing buffers)

Reply:
yeah, i actually think this line can be removed, since the PR that added the "in_isr" tracking just put it in the `MachineExternal` interrupt function. this is just a rebase artifact.